src/main/java/bio/terra/service/SamClientService.java

```diff
@@ -177,18 +178,27 @@ public void createStudyResource(AuthenticatedUserRequest userReq, UUID studyId)
         createResourceCorrectCall(samResourceApi.getApiClient(), ResourceType.STUDY.toString(), req);
     }
 
-    public String createDatasetResource(AuthenticatedUserRequest userReq, UUID datasetId) throws ApiException {
+    public String createDatasetResource(
+        AuthenticatedUserRequest userReq,
+        UUID datasetId,
+        Optional<List<String>> readersList
+    ) throws ApiException {
         CreateResourceCorrectRequest req = new CreateResourceCorrectRequest();
+
+
         req.setResourceId(datasetId.toString());
         req.addPoliciesItem(
             DataRepoRole.STEWARD.getPolicyName(),
-            createAccessPolicy(DataRepoRole.STEWARD.getRoleName(), stewardsGroupEmail));
+            createAccessPolicy(DataRepoRole.STEWARD.getRoleName(), Collections.singletonList(stewardsGroupEmail)));
         req.addPoliciesItem(
             DataRepoRole.CUSTODIAN.getPolicyName(),
-            createAccessPolicy(DataRepoRole.CUSTODIAN.getRoleName(), userReq.getEmail()));
+            createAccessPolicy(DataRepoRole.CUSTODIAN.getRoleName(), Collections.singletonList(userReq.getEmail())));
         req.addPoliciesItem(
             DataRepoRole.READER.getPolicyName(),
             new AccessPolicyMembership().roles(Collections.singletonList(DataRepoRole.READER.getRoleName())));
+        if (!readersList.isPresent()) {
+            createAccessPolicy(DataRepoRole.READER.getRoleName(), readersList.get());
```

Comment:
I think the access policy returned by this needs to be passed into `req.addPoliciesItem`. This `if` can probably be dropped and then lines 196-201 can become:

```java
req.addPoliciesItem(
    DataRepoRole.READER.getPolicyName(),
    createAccessPolicy(DataRepoRole.READER.getRoleName(), readersList.orElse(Collections.emptyList())); 
```